src/reducers/AddSiteReducer.js

```diff
@@ -74,14 +95,32 @@ export default (state = INITIAL_STATE, action) => {
             return {...state, facilitiesOption};
 
         case ADD_SITE_FIELDS_RESET:
-            return INITIAL_STATE;
+            return {...INITIAL_STATE, addSiteModalVisible: true};
```

Comment:
If we click `Reset` in the modal, return initial state of everything except for the `addSiteModalVisible` which should remain `true` to keep the modal open.